share/doc/wake/basic_wake_tutorial.md

```diff
@@ -0,0 +1,660 @@
+## Wake Tutorial
+
+### Invoking wake
+
+1. Create a wake database: `wake --init .`
+
+2. Run command for wake: `wake -vx '<function_name> <parameter_if_any>'`
+
+3. To know about various wake options run `wake --help`. `v` stands for verbosity and `x` stands for execute the expression.
+
+4. Create a file with `.wake` extension and copy the functions mentioned in this tutorial and run them using `wake` command in the terminal.  
+
+5. For more information on wake library click [here](https://sifive.github.io/wake/)
+
+### wake `Hello World`
+* If you want to display a string you can do that using the below function by writing the string to be displayed inside "" and call the function abc . Ex: `wake -x 'abc'`. This will call the function `abc` and prints `Hello World`. 
+```
+def abc = "Hello World"
+```
+
+* The other way to do this is `wake -x '"Hello World"'`
+
+### Defining Functions
```

Comment:
this may be the first time for a learner to see the wake function  and they may see this puzzling.  

for example, given, 
```
def abc = "Hello World"
def double x = 2*x
```

`wake -x 'abc' `returned  `"Hello World"`
`wake -x 'double'` (without argument) returned `<ttmp.wake:4:[16-18]>`

there got to be some explanation that the `wake -x 'double'` returned something (function reference maybe?) rather than an error. Where a user needs to go to read about more about this or is this tutorial a place to describe what is it and why is it useful?

Reply:
wikipedia says: In functional programming, functions are treated as first-class citizens, meaning that they can be bound to names (including local identifiers), `passed as arguments, and returned from other functions`, etc. 

From that `wake -x 'double' `returned a function, it seems like wake seems like a functional programming language.  Shouldn't this doc state this up front so that readers know what to expect?

Reply:
The function double takes a parameter x. So by default you have to pass a parameter to it on which the multiplication happens and output is computed